packages/service-library/src/servicelib/rabbitmq/_utils.py

```diff
@@ -51,7 +56,10 @@ async def wait_till_rabbitmq_responsive(url: str) -> bool:
 
 
 def get_rabbitmq_client_unique_name(base_name: str) -> str:
-    return f"{base_name}_{socket.gethostname()}"
+    random_string = "".join(
+        random.choice(CHARACTERS) for _ in range(_GENERATE_RANDOM_STRING_LENGTH)
+    )
+    return f"{base_name}_{socket.gethostname()}_{random_string}"
```

Comment:
Hmm, I am a bit skeptical.
The client name should not be used for defining queue names, I thought that was part of our discussion.

The client name is used to establish the connection between the service and RabbitMQ.
I think we are mixing concepts:
- client name is one thing,
- exchange names is another,
- queue names is yet another.
If we define exclusive queues, they must have a unique name, but that does not entail the client has to be then also prefixed with some weird characters.

Anyway, let's see if this fixes the issue, nevertheless the whole thing is not clean, and we should definitely review this and I would ask to create an issue for this if possible. thanks!

Reply:
Yes totally agree, the reason why it will work is that this `get_rabbitmq_client_unique_name` is overwriten with `exhange_name` of the queue is not exclusive.. yes we need to refactor this.